### ISSUE TITLE
Group principal?

### ISSUE THREAD

**Issue by Bnicke** [stnoonan/spnego-http-auth-nginx-module#74]
Is it possible to add a group somehow?
I tried `auth_gss_authorized_principal <group>@<realm>` (in several different formats), it gives me an 403, `auth_gss_authorized_principal <username>@<realm>` works.

**Comment by tbmorris** [comment]
Agreed.  The ability to control access by a single authorized group will be vastly superior to calling out each user that is authorized to access a location.

**Comment by xobot42** [comment]
Hi. Are there any news about this issue?
Will it be released? Or may be have a some trick, how do it?

**Comment by stnoonan** [comment]
No trick. To be honest, I don't even have a testbed setup for this project at this point. I'll take pull requests, but the one that would be most valuable would be adding tests :-(.

**Comment by crpb** [comment]
Hi there,

does anyone have a working example of `auth_gss_authorized_principal_regex` from 467b2a344a8eab0760cf4d8a13318714bf7f8a24 for all Users in a certain Group?

I only found [this](https://github.com/stnoonan/spnego-http-auth-nginx-module/pull/105#issuecomment-996273014) comment which never was answered.

**Comment by hawicz** [comment]
Kerberos principals have no concept of groups at all, it's really not sensible to ask about how to make `auth_gss_authorized_principal_regex` match a group.  It's the same as asking what regex would you write to check whether "John Smith" is in a group - you can't write such a regex.
The use of the phrase `<group>` in the README.md is rather misleading, and should really say something like `<instance>` instead (to match the definition of the parts of a principal, as per https://web.mit.edu/kerberos/krb5-1.5/krb5-1.5.4/doc/krb5-user/What-is-a-Kerberos-Principal_003f.html)

At most you could write a regex that matches multiple different principals, i.e. `^(usera|userb|userc)@realm`

**Comment by Alphix** [comment]
> Kerberos principals have no concept of groups at all

Indeed, this is out-of-scope of this module, you'd need to add principal-to-group-mapping (e.g. via LDAP) to your underlying application instead.

I've updated the README.md to remove references to "groups"
